### ISSUE TITLE
[Win32] Shell-TaskItem: overlay text remains after the shell is closed (Windows 11)

### ISSUE THREAD

**Issue by tmssngr** [eclipse-platform/eclipse.platform.swt#603]
**Describe the bug**
Windows 11 shows always the task items for its windows combined into one. If one shell of your multi-shell SWT application has an overlay text, it will be shown. So fine so good. But when closing the shell with the overlay text, it remains displayed.

**To Reproduce**
Please run the following snippet on Windows 11.
```java
import org.eclipse.swt.*;
import org.eclipse.swt.widgets.*;

public class TaskbarTest {

	public static void main(String[] args) {
		final Display display = new Display();

		createShell("Without taskitem", display);

		final Shell shell = createShell("With taskitem", display);
		final TaskItem item = display.getSystemTaskBar().getItem(shell);
		item.setOverlayText("abc");

		while (!display.isDisposed()) {
			if (!display.readAndDispatch()) {
				display.sleep();
			}
		}
	}

	private static Shell createShell(String text, Display display) {
		final Shell shell = new Shell(display);
		shell.setText(text);
		shell.addListener(SWT.Close, event -> {
			final Shell[] shells = display.getShells();
			if (shells.length == 1) {
				display.dispose();
			}
		});
		shell.setSize(400, 300);
		shell.open();
		return shell;
	}
}
```
You will see 2 shells, one with task item overlay, one without. Closing the one with overlay causes the task item still to show the item.

**Expected behavior**
The overlay should be hidden if the shell causing it has been closed/disposed.

**Screenshots**
![screenshot](https://user-images.githubusercontent.com/6029561/226955367-2fe95805-7032-4558-89c2-6ef019807063.png)

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [x] Windows
- - [ ] Linux
- - [ ] macOS 

2. Additional OS info (e.g. OS version, Linux Desktop, etc)
Windows 11

**Workaround (or) Additional context**
None

**Comment by deepika-u** [comment]
@tmssngr 
On windows 11 i am seeing as below : it looks to be a problem.
when both shells are open => 
![image](https://user-images.githubusercontent.com/46807064/231667396-4f366e6e-ecb4-4092-8727-147f4501af33.png)

When one shell "With taskitem" window is closed,
![image](https://user-images.githubusercontent.com/46807064/231667711-c60bb1dc-9b49-4961-84bd-3b4ecf72570f.png)

Ideally the overlay text is not set on this shell named "Without taskitem", so text "abc" should not show up.


On windows 10 i am seeing as below : it is working as expected.
when both shells are open  => 
![image](https://user-images.githubusercontent.com/46807064/231673988-b4387d79-c1f6-4690-8a7f-ca81e7c370d6.png)

When one shell "With taskitem" window is closed,
![image](https://user-images.githubusercontent.com/46807064/231674340-13e4d603-16c3-467e-9eb7-a2c02900b2cf.png)

Now the overlay text is not set on this shell named "Without taskitem", so text "abc" is not showing up. This is the correct and expected behaviour.

So on windows 11 basically when taskitems are set per window, it is considering as if it is set on all similar kind of windows and applies them to all.

**Comment by tmssngr** [comment]
@SyntevoAlex Could this be a problem of Win11 instead of SWT?

**Comment by SyntevoAlex** [comment]
To be fixed via #679
